### DISCUSSION TITLE
Higher speed

### DISCUSSION THREAD

**Discussion by MichalProkop1** [kachurovskiy/nanoels#81] [Q&A]
Hi Maxim,Thanks you for your great works . Please could you help me , Im testing version h2 . But Im use servo with MOTOR_STEPS 4000 
Im was try decrease 
#define PULSE_MIN_US round(200.0 * 200.0 / MOTOR_STEPS) // Microseconds to wait after high pulse, min.
#define PULSE_MAX_US round(2000.0 * 200.0 / MOTOR_STEPS) // Microseconds to wait after high pulse, max. Slow start. 
but still is so slowly rotation motor
It possible increase more speed of motor ? And whewre

**Comment by kachurovskiy** [comment]
Hi @MichalProkop1, I have 2 potentials solutions for you here:

1. Lower `MOTOR_STEPS` as much as you can, ideally to 200 or
2. If you're using a servo, try changing line https://github.com/kachurovskiy/nanoels/blob/main/h1/NanoEls.ino#L706 to `delayMicroseconds(5);` - this will hand over the acceleration logic to the servo driver (didn't try this myself)

Let me know if it helps. Thanks!

**Comment by MichalProkop1** [comment]
![IMG_20221111_172341](https://user-images.githubusercontent.com/117972881/201394102-4fe669cd-2083-4694-85c5-ed51db2906cc.jpg)
![IMG_20220327_110231](https://user-images.githubusercontent.com/117972881/201394133-81eb08f3-04c7-4e5a-a277-b19ec6f4ed06.jpg)
![IMG_20220327_110205](https://user-images.githubusercontent.com/117972881/201394169-c4ff6fbd-59af-4497-b5c2-0799b3f5d742.jpg)
![IMG_20220327_110236](https://user-images.githubusercontent.com/117972881/201394203-441cb745-719d-40d1-a70d-d7de4a3dc5e5.jpg)
![IMG_20220215_162919](https://user-images.githubusercontent.com/117972881/201394278-a3dbd465-f2de-4dad-8218-ec68d126e3ad.jpg)

![IMG_20220327_110242](https://user-images.githubusercontent.com/117972881/201394242-d04beb85-e0e8-4ef3-9b08-129755136068.jpg)
Hi Maxim thank you very much ,bu this    delayMicroseconds(5);   dont help ,it was the same .Im this solved by change multiplier parametr on servo. Now im have MOTOR_STEPS 400.0 and seems working . Version H1 was function with 4000  . Im sending some foto .

**Reply by kachurovskiy** [reply]
Very nice! Congratulations
